repository/src/main/java/org/alfresco/repo/action/executer/NodeSizeDetailActionExecutor.java

```diff
@@ -0,0 +1,213 @@
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.repo.action.executer;
+
+import org.alfresco.repo.action.ParameterizedItemAbstractBase;
+import org.alfresco.repo.cache.SimpleCache;
+import org.alfresco.service.cmr.action.Action;
+import org.alfresco.service.cmr.action.ParameterDefinition;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.service.cmr.repository.StoreRef;
+import org.alfresco.service.cmr.search.ResultSet;
+import org.alfresco.service.cmr.search.SearchParameters;
+import org.alfresco.service.cmr.search.SearchService;
+import org.alfresco.util.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ *   NodeSizeDetailActionExecutor
+ *   Executing Alfresco FTS Query to find size of Folder Node
+ */
+
+public class NodeSizeDetailActionExecutor extends ActionExecuterAbstractBase
+{
+
+    private static final Logger LOG = LoggerFactory.getLogger(NodeSizeDetailActionExecutor.class);
+
+    /**
+     * Action constants
+     */
+    public static final String NAME = "folder-size";
+    public static final String EXCEPTION = "Exception";
+    public static final String DEFAULT_SIZE = "default-size";
+    private static final String FIELD_FACET = "content.size";
+    private static final String FACET_QUERY = "content.size:[0 TO "+Integer.MAX_VALUE+"] \"label\": \"large\",\"group\":\"Size\"";
+    private SearchService searchService;
+    private SimpleCache<Serializable, Map<String, Object>> simpleCache;
+
+    /**
+     * Set the search service
+     *
+     * @param searchService  the search service
+     */
+    public void setSearchService(SearchService searchService)
+    {
+        this.searchService = searchService;
+    }
+
+    /**
+     * Set the simpleCache service
+     *
+     * @param simpleCache  the cache service
+     */
+    public void setSimpleCache(SimpleCache<Serializable, Map<String, Object>> simpleCache)
+    {
+        this.simpleCache = simpleCache;
+    }
+
+    /**
+     * @see ActionExecuter#execute(Action, NodeRef)
+     */
+    @Override
+    public void executeImpl(Action nodeAction, NodeRef actionedUponNodeRef)
+    {
+        Serializable serializable = nodeAction.getParameterValue(DEFAULT_SIZE);
+        int defaultItems;
+        Map<String,Object> response = new HashMap<>();
```

Comment:
As mentioned earlier, using a simple POJO to encapsulate the required information is cleaner and more maintainable than using a Map. This approach would enhance both clarity and code organisation.

Reply:
Modified
Done